firestore-bigquery-export/scripts/import/src/run.ts

```diff
@@ -0,0 +1,216 @@
+      .catch((error) => {
+        console.error(
+          "An error has occurred on the following documents, please re-run or insert the following query documents manually...",
+          JSON.stringify(serializedQuery)
+        );
+        console.error(error);
+        process.exit(1);
+      });
+  }
+
+  // Wait for all tasks to be complete.
+  while (workerPool.stats().activeTasks + workerPool.stats().pendingTasks > 0) {
+    // Return a default promise
+    // A timeout is needed here to stop infinite rechecking of workpool.stats().
+    await new Promise((resolve) => setTimeout(resolve, 150, []));
+  }
+
+  await workerPool.terminate();
+
+  console.log(`Imported ${total} documents in ${partitions} partitions.`);
+
+  console.log("---------------------------------------------------------");
+  console.log(
+    `Please see https://console.cloud.google.com/bigquery?p=${
+      config.projectId
+    }&d=${config.datasetId}&t=${config.tableId}_raw_changelog&page=table`
+  );
+  console.log("---------------------------------------------------------");
+
+  return Promise.resolve(total);
+}
+
+/**
+ * Batch import data from a collection.
+ */
+async function processCollection(config: CliConfig): Promise<number> {
+  let total = 0;
+  let batches = 0;
+  let lastDocument = null;
+  let lastBatchSize: number = config.batchSize;
+
+  while (lastBatchSize == config.batchSize) {
+    batches++;
+    let query = firebase
+      .firestore()
+      .collection(config.sourceCollectionPath)
+      .limit(config.batchSize);
+
+    if (lastDocument != null) {
+      query = query.startAfter(lastDocument);
+    }
+
+    const snapshot = await query.get();
+    const { docs } = snapshot;
+
+    const dataSink = new FirestoreBigQueryEventHistoryTracker({
+      tableId: config.tableId,
+      datasetId: config.datasetId,
+      datasetLocation: config.datasetLocation,
+    });
+
+    const rows: FirestoreDocumentChangeEvent = docs.map((document) => {
+      return {
+        timestamp: new Date(0).toISOString(),
+        operation: ChangeType.IMPORT,
+        documentName: `projects/${
+          config.projectId
+        }/databases/(default)/documents/${document.ref.path}`,
+        documentId: document.id,
+        eventId: "",
+        data: document.data(),
+      };
+    });
+
+    await dataSink.record(rows);
+
+    if (docs.length) {
+      lastDocument = docs[docs.length - 1];
+    }
+    lastBatchSize = docs.length;
+    total += docs.length;
+  }
+
+  console.log(`Imported ${total} documents in ${batches} batches.`);
+
+  return total;
+}
+
+export async function runMultiThread(): Promise<number> {
+  const config: CliConfig = await parseConfig();
+  const {
+    projectId,
+    queryCollectionGroup,
+    tableId,
+    datasetLocation,
+    datasetId,
+  } = config; // No longer needed? collectionGroup handles both types?
+
+  // Initialize Firebase
+  firebase.initializeApp({
+    credential: firebase.credential.applicationDefault(),
```

Comment:
Be sure to document that the script is using `applicationDefault()` for Firebase project. Perhaps just link to how it finds the credentials documented [here](https://cloud.google.com/docs/authentication/production).

Reply:
Additional comments as a guide for init app with applicationDefault